EIPS/eip-7870.md

```diff
@@ -0,0 +1,145 @@
+---
+eip: 7870
+title: Hardware and Bandwidth Recommendations for Validators and Full Nodes
+description: System recommendations for Validators and Full nodes
+author: Parithosh Jayanthi (@parithosh), Kevaundray Wedderburn (@kevaundray), Josh Rudolf (@jrudolf), Dankrad Feist (@dankrad), Justin Traglia (@jtraglia), Ignacio Hagopian (@jsign), George Kadianakis (@asn-d6), Fredrik Svantes (@fredriksvantes)
+discussions-to: https://ethereum-magicians.org/t/hardware-and-bandwidth-recommendations-for-full-nodes-and-validators/22675
+status: Draft
+type: Informational
+created: 2025-01-26
+---
+
+## Abstract
+
+This proposal specifies hardware and bandwidth recommendations for different types of Ethereum nodes:
+
+- **Full nodes**: Nodes that follow the tip of the chain without necessarily proposing blocks.
+- **Validators**: Split into:
+  - **Attesters**: Validators that validate and attest to blocks created by proposers.
+  - **Local block builders** (Proposers): Validators that create blocks locally and broadcast them to the network.
+
+The resource-intensive aspect for local block builders lies in creating the block and quickly broadcasting the data required for attesters to validate it in time.
+
+We note that it may be possible to run a client with less than the recommended specifications, however benchmarks and decision-making will be made with respect to these recommendations.
+
+## Motivation
+
+Clear system specifications are crucial for:
+
+- Ensuring meaningful benchmark comparisons across different client implementations.
+- Enabling informed decision-making about protocol upgrades and their resource usage implications.
+- Providing clear guidance for node operators to ensure alignment with future network requirements.
+
+Without a shared understanding of target hardware specifications:
+
+- Benchmark results lose significance due to inconsistent testing environments.
+- Decision-making becomes challenging for implementation choices, as performance characteristics are heavily hardware-dependent.
+- Network participants lack clear guidance for hardware investments.
+
+## Specification
+
+### Roles and Their Recommended Specifications
+
+Node operators typically run both an **Execution Layer (EL)** client and a **Consensus Layer (CL)** client on the same machine. The specifications below assume the combined resource usage of both.
+
+| Node Type               | Storage   | Memory | CPU Cores / Threads | **PassMark CPU Rating** | Bandwidth Download / Upload |
```

Comment:
Are all those numbers applied based on some research made? Or based on experience from discussions with stakers?

From Nethermind perspective those are super comfortable.

Also shall we distinguish EL and CL requirements to be treated separately? I saw a lot of setups where actually both are running on different machines.

Reply:
I think the average user would likely run these on the same machine? 

The rationale for them is in the linked hackmd in the comment box, we could not include all of those links so the EIP has been somewhat watered down